research/H1/FPN.md

```diff
@@ -0,0 +1,106 @@
+# Label: H1, Preamble: "FPN"
+
+Flight Plan
+
+## Format
+
+The message begins with `FPN/` followed by a status. After that, it is a list of key/value pairs delimited by `:`. The final 4 characters are a checksum (algorithm unknown).
+
+### Known Statuses
+
+key | value
+---|-----
+RI | Route Inactive
+FNXXXX | Flight Number XXXX
+
+### Known Keys
+
+key | value
+---|-----
+A  | Arrival Procedure
+AA | Arrival Airport
+AP | Approach Procedure (?)
+CR | Current Route (?)
+D  | Departure Procedure
+DA | Departure Airport
+F  | First Waypoint
+R  | Departure Runway
+
+### First Waypoint
+There can be multiple First Waypoints in a message. I believe they are related to the prior keypair. see [Landing](#landing)
+
+### Waypoints format
+The use of the waypoints is designated by the prior key/value pair. 
+Waypoints can either be delimiated by `..`, `.` or `,`. The exact meaning is currently unclear. Current guess is as follows (with incorrect terminology)
```

Comment:
For the route, it looks like it may be something like `[NAVAID/WAYPOINT],[GPS DDM]` delimited by `.`. The GPS appears to be Degree Decimal Minute format (somewhat confirmed if you plug in the navaid/waypoint into [opennav](https://opennav.com) and look at the coordinates).

I have another example (assuming in the format) that shows that the delimiter is `.` and not `..`:
<pre>
FPN/FNAAL2338/RP:DA:KPVD:AA:KCLT:CR:PVDCLT01:R:05O(36R):A:CHSLY5.LYH:AP:ILS36R..MXE,N39551W075402.Q75.GVE..LYH,N37153W079142A557
</pre>
Notice `MXE,N39551W075402.Q75.GVE`, this is a `[VORTAC],[GPS COORDS].[ROUTE].[NEXT VORTAC]`. However, it is odd that there is no GPS for the `NEXT VORTAC`

Reply:
Yea, i was thinking it might be`..{waypoint}[,{position}]` but the coords seemed too far off. I guess just to make sure, when you say DDM `N39551W075402` would convert to `39.551,-075.402`, correct?

I'll add this message to the docs, which scraps the waypoint name and just goes from position to position

```
FPN/RI:DA:KMEM:AA:EGSS:F:HUMMS,N36214W088591..PXV,N37557W087457..ROD,N
40173W084026..SLT,N41308W077582.J190..RKA,N42280W075144..MIILS,N46524W
067029.N279A..TUDEP,N51100W053140..N52000W050000..N53000W040000..N5400
0W030000..N54000W020000..DOGAL,N54000W015000..BEXET,N54000W014000..BAK
UR,N52145W005408.UN546..STU,N51597W005024.UP2..NUMPO,N51366W003170.Y3.
.BEDEK,N51223W001335CE78
```

`..` vs `.` really bothers me... i'd love to figure out what it means

Reply:
So from your example, let's take `ROD`. The GPS coords per OpenNav is  40° 17' 16.06" N, 84° 2' 35.31" W (see https://opennav.com/navaid/US/ROD/ROSEWOOD ). It sort matches the `N40173W084026` after it: `N 40 17.3, W 84 02.6`. 

Regarding, `..` and `.`, the same example unfortunately just busted through my assumptions with `STU,N51597W005024.UP2..NUMPO,N51366W003170`...